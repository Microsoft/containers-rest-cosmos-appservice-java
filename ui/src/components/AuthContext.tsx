import { createContext } from 'react'

export interface IAuthContextValues {
  accessToken: string,
  authResponse: string,
  handleAuth: () => void,
  setAuthResponse: (msg?: string) => void,
}

// Default values and enforce use of interface
const defaultAuthContextValue: IAuthContextValues = {
  accessToken: null,
  authResponse: null,
  handleAuth: () => null,
  setAuthResponse: () => null,
}

export const AuthContext = createContext<IAuthContextValues>(defaultAuthContextValue)
// exporting the Provider and Consumer components for more specific imports
export const AuthProvider = AuthContext.Provider
export const AuthConsumer = AuthContext.Consumer
