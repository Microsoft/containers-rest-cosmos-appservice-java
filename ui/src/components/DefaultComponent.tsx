import * as React from 'react'

export interface IDefaultComponentProps {
    default: boolean,
}

export interface IDefaultComponentState {
    deerSrc: string,
}

export class DefaultComponent extends React.Component<IDefaultComponentProps> {
    public state = {
        deerSrc: null,
    }

    public componentWillMount() {
        this.setState({ deerSrc: 'https://i.gifer.com/KG8.gif' })
    }

    public render() {
        return (
            <div className='default-component cover'>
                <h1>
                    Oh Deer :(
                <br />
                    <small>
                        Error 404
                    </small>
                </h1>
                <p className='lead'>
                    The requested resource could not be found but may be available again in the future.
            </p>
                <img className='default-page-image' src={this.state.deerSrc} />
            </div>
        )
    }
}
