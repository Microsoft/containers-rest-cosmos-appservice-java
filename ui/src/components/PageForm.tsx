import * as React from 'react'

export interface IPageFormProps {
  inputTitle: string,
  inputPlaceholder: string,
  onInputChange: (event) => void,
  onSubmitClick: (event) => void,
}

export function PageForm(props: IPageFormProps) {
  return (
    <form className='form'>
      <label>
        <span className='form-field-pre-text'>{props.inputTitle}</span>
        <input
          className='form-field-input'
          type='text'
          name='id'
          onChange={props.onInputChange}
          placeholder={props.inputPlaceholder}
        />
        <span className='form-field-sub-text'>Leave empty for random sample</span>
      </label>
      <input
        className='form-field-submit'
        type='submit'
        value='Search'
        onClick={props.onSubmitClick}
      />
    </form>
  )
}
