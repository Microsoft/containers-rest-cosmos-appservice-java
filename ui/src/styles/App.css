@import url('https://fonts.googleapis.com/css?family=Open+Sans');

body {
  font-family: 'Open Sans', Helvetica, sans-serif;
  padding: 0;
  margin: 0;
}

.app-container {
  display: flex;
  flex-direction: column;
  height: 100%;
  margin-bottom: 20px;
}

.page-container {
  flex-grow: 1;
  display: flex;
  flex-direction: row;
  padding: 0 10px;
  justify-content: space-around;
}

.page-container--home {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.form-container {
  width: 300px;
}

.form-description {
  text-align: justify;
}

.form {
  display: flex;
  flex-direction: column;
}

.form-field-input {
  display: block;
  width: 300px;
  border-radius: 5px;
  font-size: 1rem;
  -webkit-box-shadow: inset 2px 2px 2px 0px #dddddd;
  -moz-box-shadow: inset 2px 2px 2px 0px #dddddd;
  box-shadow: inset 2px 2px 2px 0px #dddddd;
  border: 1px solid #ccc;
  padding: 7px 10px;
  box-sizing: border-box;
}

.form-field-submit {
  width: 300px;
  margin-top: 20px;
  background-color: rgb(50, 146, 202);
  border: 1px solid rgb(50, 146, 202);
  border-radius: 5px;
  color: #FFF;
  font-size: 1.2rem;
  padding: 10px 0;
}

.results-container {
  padding: 20px;
  width: max-content;
  max-width: 50%;
}

.results-view {
  overflow-x: scroll;
  border: 1px solid #ccc;
  background: rgba(0, 0, 0, 0.1);
  box-shadow: inset 1px 1px 3px rgba(0, 0, 0, 0.2);
  padding: 20px;
}

.home-title {
  text-align: center;
}

.home-img {
  width: 250px;
  margin: auto;
}

.center {
  margin: auto;
  width: 50%;
}

.auth-message-container {
  background-color: #ccc;
  padding: 5px 15px;
}
.auth-message {
  color: rgb(221, 26, 26)
}

.auth-message-close-button {
  color: rgb(221, 26, 26);
  font-weight: 700;
  padding: 5px;
  margin-right: 20px;
  border: none;
  cursor: pointer;
}

.login-button {
  border: none;
  cursor: pointer;
  background-color: rgb(50, 146, 202);
}

.default-page-image {
  padding-left: 33%;
  width: 400px;
}
