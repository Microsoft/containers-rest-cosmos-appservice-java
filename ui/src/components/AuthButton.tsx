import * as React from 'react'

import { AuthContext } from './AuthContext'

export class AuthButton extends React.Component {
    public static contextType = AuthContext
    public render() {
        return (
            <button
                onClick={this.context.handleAuth}
                className='login-button'
            >
                {this.context.accessToken ? 'Log Out' : 'Log In'}
            </button>
        )
    }
}
