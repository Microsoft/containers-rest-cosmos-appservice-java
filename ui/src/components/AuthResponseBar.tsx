import * as React from 'react'

import { AuthContext } from './AuthContext'

export class AuthResponseBar extends React.Component {
  public static contextType = AuthContext
  public render() {
    return this.context.authResponse && (
      <div className='auth-message-container'>
        <button className='auth-message-close-button' onClick={() => this.context.setAuthResponse(null)}>X</button>
        <span className='auth-message'>{this.context.authResponse}</span>
      </div>
    )
  }
}
