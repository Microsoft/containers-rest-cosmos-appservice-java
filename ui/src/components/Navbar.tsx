import { Link } from '@reach/router'
import * as React from 'react'

import { AuthButton } from './AuthButton'

import './../styles/Navbar.css'

const isActive = ({ isCurrent }) => {
    return {
        className: isCurrent ? 'nav-link active' : 'nav-link',
    }
}

export class Navbar extends React.Component {
    public render() {
        return (
            <nav className='nav-container'>
                <span className='nav-title'>
                    Project Jackson
                </span>
                <div className='nav-link-container'>
                    <Link getProps={isActive} to='/'>Home</Link>
                    <Link getProps={isActive} to='/people'>People</Link>
                    <Link getProps={isActive} to='/titles'>Titles</Link>
                    <AuthButton />
                </div>
            </nav>
        )
    }
}
