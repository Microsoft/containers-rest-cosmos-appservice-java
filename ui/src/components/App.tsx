import { Router } from '@reach/router'
import * as Msal from 'msal'
import * as React from 'react'

import { AuthProvider } from './AuthContext'
import { AuthResponseBar } from './AuthResponseBar'
import { DefaultComponent } from './DefaultComponent'
import { Home } from './Home'
import { Navbar } from './Navbar'
import { Person } from './Person'
import { PrivateRoute } from './PrivateRoute'
import { Title } from './Title'

import './../styles/App.css'

const appConfig = {
    clientID: WEBPACK_PROP_AAD_CLIENT_ID,
}

// initialize the UserAgentApplication globally so popup and iframe can run in the background
const userAgentApp = appConfig.clientID && new Msal.UserAgentApplication(appConfig.clientID, null, null)
// short circuit userAgentApp. If clientID is an empty string so is userAgentApp

export class App extends React.Component {

    // App is still responsible for managing the auth state
    // it uses the AuthContext to share this with other aspects of the UI
    // including the navbar, private route component, and the pages themselves
    public state = {
        accessToken: null,
        authResponse: null,
    }

    public handleAuth = async () => {
        if (!appConfig.clientID) {
            if (this.state.accessToken !== null) {
                this.setState({ accessToken: null })
            } else {
                // tslint:disable-next-line no-console max-line-length
                console.warn('AAD Client ID has not been configured. If you are currently in production mode, see the \'deploy\' documentation for details on how to fix this.')
                this.setState({ accessToken: '' })
            }
        } else {
            let accessToken = null
            try {
                if (this.state.accessToken) {
                    // log out
                    await userAgentApp.logout()
                } else {
                    // log in
                    if (!appConfig.clientID) {
                        throw new Error(
                            'AAD Client ID has not been configured. See the \'deploy\' documentation for more details.')
                    }
                    const graphScopes = [appConfig.clientID]
                    await userAgentApp.loginPopup(graphScopes)
                    accessToken = await userAgentApp.acquireTokenSilent(graphScopes,
                        'https://login.microsoftonline.com/microsoft.onmicrosoft.com')
                }
                this.setState({ accessToken })
            } catch (err) {
                this.setAuthResponse(err.toString())
            }
        }
    }

    public setAuthResponse = (msg: string) => {
        this.setState({ authResponse: msg })
    }

    public render() {
        // Implementing the authprovider at app root to share the auth state
        // with all internal components (if they subscribe to it)
        // by linking the accessToken to the app state we can be certain the
        // context will always update and propogate the value to subscribed nodes
        return (
            <AuthProvider value={{
                accessToken: this.state.accessToken,
                authResponse: this.state.authResponse,
                handleAuth: this.handleAuth,
                setAuthResponse: this.setAuthResponse,
            }}>
                <div className='app-container'>
                    <Navbar />
                    <AuthResponseBar />
                    <Router>
                        <Home path='/' />
                        <PrivateRoute as={Person} path='/people' />
                        <PrivateRoute as={Title} path='/titles' />
                        <DefaultComponent default />
                    </Router>
                </div>
            </AuthProvider>
        )
    }
}
