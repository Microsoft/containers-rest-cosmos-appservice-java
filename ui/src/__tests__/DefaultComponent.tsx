import {
    createHistory,
    createMemorySource,
    LocationProvider,
} from '@reach/router'
import { render } from 'enzyme'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { App } from '../components/App'
import { DefaultComponent } from '../components/DefaultComponent'

describe('<DefaultComponent />', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<DefaultComponent default />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    describe('It is the default 404 page', () => {
        test('It finds the page if user tries a nonsense path', () => {
            const badPath = '/awefwaef'
            const source = createMemorySource(badPath)
            const hist = createHistory(source)
            const r = render(
                <LocationProvider history={hist}>
                    <App />
                </LocationProvider>,
                )

            expect(r.find('.default-component')).toHaveLength(1)
            expect(r.find('.default-component1')).toHaveLength(0)
        })

        test('It doesn\'t find the page if user tries a valid path', () => {
            const goodPath = '/titles'
            const source = createMemorySource(goodPath)
            const hist = createHistory(source)
            const r = render(
                <LocationProvider history={hist}>
                    <App />
                </LocationProvider>,
                )

            expect(r.find('.default-component')).toHaveLength(0)
            })
        })
    })
