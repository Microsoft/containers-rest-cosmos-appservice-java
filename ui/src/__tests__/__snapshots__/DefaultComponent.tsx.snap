// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultComponent /> renders correctly 1`] = `
<div
  className="default-component cover"
>
  <h1>
    Oh Deer :(
    <br />
    <small>
      Error 404
    </small>
  </h1>
  <p
    className="lead"
  >
    The requested resource could not be found but may be available again in the future.
  </p>
  <img
    className="default-page-image"
    src="https://i.gifer.com/KG8.gif"
  />
</div>
`;
